### ISSUE TITLE
Better textures

### ISSUE THREAD

**Issue by raymoo** [DreamBlocks/DreamBlocks#71]
16x16 material-ey textures do not work ok in 2D like they do in 3D, because there is nothing to give them depth.

Some suggestions:
 - Shape nodes to look like what they are, not just flat squares of texture. For example, tree leaves could be made solid (touching the edges) only where they connect to other tree leaves, and otherwise should have some transparent margin where they have a bushy outline (because at the edges the shape of the leaves will affect the outline).
 - Switch to a higher resolution. It may be difficult to give the details needed to give the sprites more depth at 16x16 (but feel free to ignore this if you can do it)
 - Make nodes smaller. Nodes might be a quarter of their current size, or even less, allowing some detail and shape to come from the placement of nodes so that less (but still something) needs to be done in the textures. Player digging and placement would need to be modified to work with a smaller node size, maybe with digging targeting all the nodes in a radius around the cursor, and placement placing multiple nodes at a time (configurable by user?)

**Comment by AKryukov92** [comment]
Thank you for your feedback.
* Different look of nodes, depending on adjacent blocks seems like good feature. Though this look needs to be drawn and configured. I have seen such feature in Battle for Wesnoth and description of aligned look seems to be very complex.
* I don't know why current resolution has been chosen. So I think it is a subject of discussion with @LordStephen77. It is ok as it is now for me.
* Smaller nodes with some special digging mechanics seems like additional complexity with little gain.

Could you draw prototype of special look of leaves, adjacent to wood?
Or maybe more detailed prototype of player sprite?

**Comment by raymoo** [comment]
I'm not that good of an artist, all I can do is say that I think the current sprites look bad.

**Comment by RodrigueLemaire** [comment]
that true, but ithink they are only temporary as @tobyplowy did good ones.

**Comment by tobyplowy** [comment]
@azekillDIABLO thanks also I'm not sure if my textures will be used it all depends on what @LordStephen77 thinks

**Comment by AKryukov92** [comment]
@tobyplowy You can just change textures and commit them. No additional coding is required.
I did an experiment with changing sprite of "dirt".

Dirt before
![image](https://user-images.githubusercontent.com/24266616/28329712-954dab4c-6c0c-11e7-88ed-978387a2722b.png)
Dirt after
![image](https://user-images.githubusercontent.com/24266616/28329652-5f6ad84c-6c0c-11e7-86ee-e874bbeb4969.png)
I will not commit this change. I made it only to prove that no coding is required.

**Comment by tobyplowy** [comment]
> No additional coding is required.

Yeh what about the random Textures :/

**Comment by AKryukov92** [comment]
Hey. Noone has menthioned random textures before. Why would you want to do them?

**Comment by tobyplowy** [comment]
> Noone has menthioned random textures before 

yes I did in the first project same issue I posted the dirt textures I made
(I can't find it back @LordStephen77 probably removed it)

> Why would you want to do them?

Because a game like this looks terrible without random textures

**Comment by AKryukov92** [comment]
@tobyplowy Make and commit at least two examples of random textures for any tile and I will implement all required code in next two days after commit.
There are some problems in implementing inventory layout, but this thing much easier.

**Comment by tobyplowy** [comment]
@AKryukov92 i already did on the other page (I made dirt and stone) can you find that issue or is it gone?
(I probably still have them on my PC but im not sure) also if I need to make new tiles I will have to have a system to fallow perhaps we can design it together (that would be useful for me bdw I can only give ideas I can't program)

**Comment by AKryukov92** [comment]
@tobyplowy Please be more autonomous. I don't know where to search for your dirt and stone.
Find your images, commit them to repository and make pull request yourself.

What do you mean by "system to fallow"? I'm little sceptical about it.

Main point of this thread: There should be more detailed textures.
Can you just add more details to current images without need of introducing new systems?

**Comment by tobyplowy** [comment]
grass
![grass](https://user-images.githubusercontent.com/16853304/28385776-990b4fec-6cc9-11e7-931d-a2078239b866.png)
dirt
![dirt](https://user-images.githubusercontent.com/16853304/28385780-9930c0e2-6cc9-11e7-8b28-989a3348e25c.png)
stone
![stone](https://user-images.githubusercontent.com/16853304/28385778-9919b7ee-6cc9-11e7-8675-74db727095b8.png)
cobblestone
![cobblestone](https://user-images.githubusercontent.com/16853304/28385779-992626c8-6cc9-11e7-9312-bf1b43360dc5.png)

**Comment by tobyplowy** [comment]
> Can you just add more details to current images without need of introducing new systems?

Yes but it's too repetitive and boring that's why there should be randomized textures bdw I posted them above

**Comment by AKryukov92** [comment]
@tobyplowy  Don't post here your work. Check that game looks good, commit and then make pull request.
I will not do anything with sprites you attach to this thread.
It is up to you to add them to the repository in correct format.

**Comment by tobyplowy** [comment]
@AKryukov92 i cant program so i cant properly add them

**Comment by AKryukov92** [comment]
@tobyplowy  You don't need to code. just replace existing images in repository. Or add them, if they are variants of random textures.

**Comment by tobyplowy** [comment]
@AKryukov92 yes but how do i get the random textures and tiling

**Comment by AKryukov92** [comment]
@tobyplowy as I said before. I'll implement randomisation of tile textures in two days after you commit them.
I said nothing about tiling, though. What is it?

**Comment by tobyplowy** [comment]
> What is it?

it makes the edges nice
![tile example](https://user-images.githubusercontent.com/16853304/28386923-776ed882-6ccd-11e7-98e4-aee45c2be330.png)

**Comment by tobyplowy** [comment]
> after you commit them.

so do i just replace the curent ones with this (that will probably brake the game if theres no prober system)
![grass](https://user-images.githubusercontent.com/16853304/28386983-a679fbac-6ccd-11e7-9bab-acdeb60cb4ad.png)

**Comment by tobyplowy** [comment]
@AKryukov92 there https://github.com/DreamBlocks/DreamBlocks/pull/73

**Comment by AKryukov92** [comment]
Each tile image should be in separate file with similar name.
Like dirt.png, dirt1.png and so on.
Check the game look after your changes. It should not change or become better.

**Comment by tobyplowy** [comment]
@AKryukov92 is this good? https://github.com/DreamBlocks/DreamBlocks/pull/74

**Comment by AKryukov92** [comment]
No

**Comment by tobyplowy** [comment]
@AKryukov92 Ok whatever I'm very tred and I need sleep it's hard to think straight :/

**Comment by RodrigueLemaire** [comment]
@tobyplowy don't care it's nice work! tiling is very nice!

**Comment by tobyplowy** [comment]
@LordStephen77 what do you think?
